src/goto-inspect/goto_inspect_parse_options.cpp

```diff
@@ -0,0 +1,99 @@
+// Author: Fotis Koutoulakis for Diffblue Ltd.
+
+#include "goto_inspect_parse_options.h"
+
+#include <util/config.h>
+#include <util/exception_utils.h>
+#include <util/exit_codes.h>
+#include <util/invariant.h>
+#include <util/make_unique.h>
+#include <util/version.h>
+
+#include <goto-programs/goto_model.h>
+#include <goto-programs/read_goto_binary.h>
+#include <goto-programs/show_goto_functions.h>
+
+#include <iostream>
+
+int goto_inspect_parse_optionst::doit()
+{
+  if(cmdline.isset("version"))
+  {
+    std::cout << CBMC_VERSION << '\n';
+    return CPROVER_EXIT_SUCCESS;
+  }
+
+  // Before we do anything else, positional arguments needs to be > 1.
+  // (i.e. a filename has been given).
+  if(cmdline.args.size() < 1)
+  {
+    help();
+    throw invalid_command_line_argument_exceptiont{
+      "failed to supply a goto-binary name",
+      "<input goto-binary> <inspection-option>"};
+  }
+
+  // This just sets up the defaults (and would interpret options such as --32).
+  config.set(cmdline);
+
+  // Normally we would register language front-ends here but as goto-harness
+  // only works on goto binaries, we don't need to
+
+  auto binary_filename = cmdline.args[0];
+
+  // Read goto binary into goto-model
+  auto read_goto_binary_result =
+    read_goto_binary(binary_filename, ui_message_handler);
+  if(!read_goto_binary_result.has_value())
+  {
+    throw deserialization_exceptiont{
+      "failed to read goto program from file '" + binary_filename + "'"};
+  }
+  auto goto_model = std::move(read_goto_binary_result.value());
+
+  // This has to be called after the defaults are set up (as per the
+  // config.set(cmdline) above) otherwise, e.g. the architecture specification
+  // will be unknown.
+  config.set_from_symbol_table(goto_model.symbol_table);
+
+  if(
+    cmdline.isset("show-goto-functions") ||
+    cmdline.isset("list-goto-functions"))
```

Comment:
This alias is not configured in GOTO_INSPECT_OPTIONS.

Reply:
Removed.